app/models/spree/order_decorator.rb

```diff
@@ -263,7 +263,15 @@ def set_variant_attributes(variant, attributes)
   end
 
   def line_item_variants
-    line_items.map(&:variant)
+    if line_items.loaded?
+      line_items.map(&:variant)
+    else
+      line_items.includes(:variant).map(&:variant)
+    end
+  end
```

Comment:
I really feel like we are reinventing AR. Would you mind checking where is this `#line_items_variants` used?

Reply:
It's used once in one of our services, in: `OrderCycleDistributedVariants#unavailable_line_items`, and a couple of times in some specs, apparently just as a shortcut.

Reply:
can we create an issue to revisit these? I would reuse the association in those cases.